nixos/modules/services/networking/wasabibackend.nix

```diff
@@ -0,0 +1,145 @@
+        default = "/var/lib/wasabibackend";
+        description = "The data directory for the Wasabi backend node.";
+      };
+
+			customConfigFile = mkOption {
+				type = types.nullOr types.path;
+				default = null;
+				description = "Defines the path to a custom configuration file that is copied to the user's directory. Overrites any config options.";
+			};
+
+      network = mkOption {
+        type = types.enum [ "mainnet" "testnet" "regtest" ];
+        default = "mainnet";
+        description = "The network to use for the Wasabi backend service.";
+      };
+
+			endpoint = {
+				ip = mkOption {
+					type = types.str;
+					default = "127.0.0.1";
+					description = "IP address for P2P connection to bitcoind.";
+				};				
+				port = mkOption {
+					type = types.port;
+					default = 18333;
+					description = "Port for P2P connection to bitcoind.";
+				};
+			};
+
+      rpc = {
+				ip = mkOption {
+					type = types.str;
+					default = "127.0.0.1";
+					description = "IP address for RPC connection to bitcoind.";
+				};
+
+				port = mkOption {
+					type = types.port;
+					default = 18332;
+					description = "Port for RPC connection to bitcoind.";
+				};
+
+        user = mkOption {
+          type = types.str;
+          default = "bitcoin";
+          description = "RPC user for the bitcoin endpoint.";
+        };
+
+        password = mkOption {
+          type = types.str;
+          default = "password";
+          description = "RPC password for the bitcoin endpoint. Warning: this is stored in cleartext in the Nix store! Use <literal>configFile</literal> if needed.";
+        };
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "wasabibackend";
+        description = "The user as which to run the wasabibackend node.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = cfg.user;
+        description = "The group as which to run the wasabibackend node.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd.tmpfiles.rules = [
+      "d '${cfg.dataDir}' 0770 '${cfg.user}' '${cfg.group}' - -"
+    ];
+
+    systemd.services.wasabibackend = {
+      description = "wasabibackend server";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network-online.target" ];
+			environment = {
+				DOTNET_PRINT_TELEMETRY_MESSAGE = "false";
+				DOTNET_CLI_TELEMETRY_OPTOUT = "true";
+			};
+      preStart = ''
+        mkdir -p ${cfg.dataDir}/.walletwasabi/backend
+				${if cfg.customConfigFile != null then ''
+					cp -v ${cfg.customConfigFile} ${cfg.dataDir}/.walletwasabi/backend/Config.json
+				'' else ''
+					cp -v ${configFile} ${cfg.dataDir}/.walletwasabi/backend/Config.json	
+				''}
+				chmod ug+w ${cfg.dataDir}/.walletwasabi/backend/Config.json
+      '';
+      serviceConfig = {
+        User = cfg.user;
+        Group = cfg.group;
+        ExecStart = "${pkgs.wasabibackend}/bin/WasabiBackend";
+      };
```

Comment:
The daemon looks like it doesn't need to interact with system much, would be nice to include sandboxing options like `ProtectSystem` and perhaps `DynamicUser`.

Reply:
I tried with `DynamicUser` but sadly the Wasabi backend did not like that, as it is checking for a system user. This piece of software has a long way to go. For example it requires write access to the config file etc. I will check what other enforcement it can work with.